x-pack/plugins/observability_solution/apm/server/routes/assistant_functions/get_observability_alert_details_context/get_context_message.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { isEmpty } from 'lodash';
+
+export function getContextMessage({
+  serviceName,
+  serviceEnvironment,
+  serviceSummary,
+  downstreamDependencies,
+  logCategories,
+  serviceChangePoints,
+  exitSpanChangePoints,
+  anomalies,
+}: any): string {
+  const obsAlertContext = `${
+    !isEmpty(serviceSummary)
+      ? `Metadata for the service where the alert occurred:
+${JSON.stringify(serviceSummary, null, 2)}`
+      : ''
+  }
```

Comment:
I originally wrote this but seeing it now, perhaps it's easier to read and maintain if we do something like this:

```ts
  const prompt: Array<[string, Record<string, unknown>]> = [];

  if (!isEmpty(serviceSummary)) {
    prompt.push(['Metadata for the service where the alert occurred', serviceSummary]);
  }

  // ...other data

  const promptAsString = prompt
    .map(([dataDescription, data]) => `${dataDescription}:\n${JSON.stringify(data, null, 2)}`)
    .join('\n\n');
```

Reply:
Another approach is to register data as semi-structured:

```ts
  const prompt: Array<[string, Record<string, unknown>]> = [];

  if (!isEmpty(serviceSummary)) {
    prompt.push(['Metadata for the service where the alert occurred', serviceSummary]);
  } 

  return prompt;
```

Then leaving it up to the handler the observability plugin to format it. That would make it easier to add the query param  (`json=true`)  for getting structured output that we need in api tests